### PR TITLE
Refactor CircleCI test reporting

### PR DESCRIPTION
**Pull request by taymoork2**
## Description

With the move to a bash script for testing packages, testing was improperly exiting with a code of `0`, even if a test reported a failure. This fixes that and removes the need for the other script to look through xml files to see if something had failed or not.

Fixes [SPARK-106753](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-106753)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/plugin-phone/src/index.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ import {registerPlugin} from '@webex/webex-core';
 import Phone from './phone';
 import config from './config';
 
+console.log('hello');
```
Comment (mwegman) [inline packages/node_modules/@webex/plugin-phone/src/index.js]: Remove this commit.

File: tooling/lib/test/karma.js [hunk 2]
```diff
@@ -25,31 +25,22 @@ exports.test = async function test(options, packageName, files) {
   }
 
   if (options.xunit || process.env.COVERAGE || process.env.CIRCLECI || process.env.CI) {
-    for (let i = 0; i < 3; i += 1) {
```
Comment (mwegman) [inline tooling/lib/test/karma.js]: Why did you remove the 'retry 3 times'?

**Comment by adamweeks** [comment]
Can you provide a link to a circle job that you verified this config fails as expected?

File: packages/node_modules/@webex/plugin-phone/src/index.js (hunk 1 above)
Comment (taymoork2) [reply packages/node_modules/@webex/plugin-phone/src/index.js]: This commit is just here for circle to test something so im gonna remove it once i get the results im looking for
There's a script that diff's against the latest tag for changes in the packages, if there aren't any changes it's not gonna run tests

**Comment by taymoork2** [comment]
My bash wasn't good enough and the npm script was reporting with an exit code 0 even with a failures
So back to parsing xml files

Link to verified results

File: tooling/lib/test/karma.js (hunk 2 above)
Comment (taymoork2) [reply tooling/lib/test/karma.js]: This is old code I'm guessing that was setup for Jenkins and relative flakiness from back when it was probably necessary to retry.
Results are more reliable now and could give the possibility of testing against more "trickle-down" packages instead of only modified packages

File: .circleci/config.yml [hunk 3]
```diff
@@ -134,13 +134,11 @@ jobs:
       - run:
           name: 'Check if any unit tests failed'
           command: |
-            FAILURES=$(find reports -iname 'webex*.xml' -print0 | xargs -0 sed -n 's/.*failures=\"\([^\"]*\).*/\1/p' | xargs)
-            ERRORS=$(find reports -iname 'webex*.xml' -print0 | xargs -0 sed -n 's/.*errors=\"\([^\"]*\).*/\1/p' | xargs)
```
Comment (adamweeks) [inline .circleci/config.yml]: Why are we no longer checking ERRORS?

Comment (taymoork2) [reply .circleci/config.yml]: It was a while ago, but one of Arun's PR's had passed all the tests but the  job itself failed because mocha/karma had reported an error. It didn't report what the error was so to me it was just a false positive for a passing test suite and an error is more on the runner side than the test side
